### DISCUSSION TITLE
With new `Link` component how can we create a navigation item that will always take us to a section of a particular page (it should be accesible from other pages too). Also how one could dynamically highlight it (`usePathname()` and `useLayoutSegment()` cannot tell us what id is currently in the URL. Could someone please guide me on how to achieve such a thing in nextjs 13? I cannot infer that from the docs

### DISCUSSION THREAD

**Discussion by joehagedorn** [vercel/next.js#45346] [Help]
With new `Link` component how can we create a navigation item that will always take us to a section of a particular page (it should be accesible from other pages too). Also how one could dynamically highlight it (`usePathname()` and `useLayoutSegment()` cannot tell us what id is currently in the URL. Could someone please guide me on how to achieve such a thing in nextjs 13? I cannot infer that from the docs

_Originally posted by @krzysztofMlczk in https://github.com/vercel/next.js/discussions/41745#discussioncomment-4745053_

**Comment by joehagedorn** [comment]
The Link component seems to be a work in progress but could indeed resolve the typical id segment routes more efficiently in server side renders?

**Comment by GabenGar** [comment]
Fragments are never resolved on server if only because they require the knowledge of the document's rendered layout to work. Something which they obviously can't know.
Not to mention you have to ensure stable layout for these to work in the first place, since the scroll event from fragment fires only once the moment html is parsed. So if you have some client-side logic pushing layout down, it will leave the user in confused state.
